cmd/gitserver/server/vcs_syncer_jvm_packages.go

```diff
@@ -177,6 +180,34 @@ func (s *JVMPackagesSyncer) packageDependencies(ctx context.Context, repoUrlPath
 		}
 	}
 
+	dbDeps, err := s.DBStore.GetJVMDependencyRepos(ctx, dbstore.GetJVMDependencyReposOpts{
+		ArtifactName: repoUrlPath,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	for _, dep := range dbDeps {
+		parsedModule, err := reposource.ParseMavenModule(dep.Module)
+		if err != nil {
+			log15.Warn("error parsing maven module", "error", err, "module", dep.Module)
+			continue
+		}
+		fullDependencyString := reposource.MavenDependency{
+			MavenModule: parsedModule,
+			Version:     dep.Version,
+		}.CoursierSyntax()
+		if module.MatchesDependencyString(fullDependencyString) {
```

Comment:
Nit: we don't need `fullDependencyString` to test if the dependency matches
```suggestion
		if module == parsedModule {
```